src/libs/CategoryUtils.ts

```diff
@@ -68,4 +68,14 @@ function getCategoryDefaultTaxRate(expenseRules: ExpenseRule[], categoryName: st
     return categoryDefaultTaxRate;
 }
 
-export {formatDefaultTaxRateText, formatRequireReceiptsOverText, getCategoryApproverRule, getCategoryExpenseRule, getCategoryDefaultTaxRate};
+function updateCategoryInMccGroup(mccGroups: Record<string, MccGroup>, oldCategoryName: string, newCategoryName: string) {
+    const updatedGroups: Record<string, MccGroup> = {};
+
+    for (const [key, group] of Object.entries(mccGroups || {})) {
+        updatedGroups[key] = group.category === oldCategoryName ? {...group, category: newCategoryName, pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE} : group;
```

Comment:
@youssef-lr I have added a condition to prevent unnecessary category updates by checking name equality at [L72](https://github.com/Expensify/App/pull/51390/files#diff-5f4e5be04b2ed93b93d55a6b8d6f4ecc522639609a1c2f0454be3fd2d82e7dc1R72).